File: implement-shell-tools/ls/ls.js
const fs = require("fs");

const args = process.argv.slice(2);

let flagA = false;
let dir = ".";

for (const arg of args) {
  if (arg === "-a") flagA = true;
  else if (arg === "-1") continue; // siempre listamos uno por línea
  else dir = arg;
}

const entries = fs.readdirSync(dir);

for (const entry of entries.sort()) {
  if (!flagA && entry.startsWith(".")) continue;
  console.log(entry);
}

File: implement-shell-tools/ls/ls.py
import sys
import os

args = sys.argv[1:]

flag_a = False
directory = "."

for arg in args:
    if arg == "-a":
        flag_a = True
    elif arg == "-1":
        continue
    else:
        directory = arg

entries = sorted(os.listdir(directory))

for entry in entries:
    if not flag_a and entry.startswith("."):
        continue
    print(entry)